prov/peer/src/peer_cq.c

```diff
@@ -89,17 +95,36 @@ int coll_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
 
 	cq->peer_cq = peer_context->cq;
 
-	ret = ofi_cq_init(&coll_prov, domain, attr, &cq->util_cq, &ofi_cq_progress,
-			  context);
+	ret = ofi_cq_init(provider, domain, attr, &cq->util_cq, 
+		&ofi_cq_progress, NULL);
 	if (ret)
 		goto err;
 
 	*cq_fid = &cq->util_cq.cq_fid;
-	(*cq_fid)->fid.ops = &coll_cq_fi_ops;
-	(*cq_fid)->ops = &coll_cq_ops;
+	(*cq_fid)->fid.ops = &peer_cq_fi_ops;
+	(*cq_fid)->ops = &peer_cq_ops;
 	return 0;
 
 err:
 	free(cq);
 	return ret;
 }
+
+int ofi_peer_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
+		struct fid_cq **cq_fid, void *context)
+{
+	struct fi_peer_cq_context *peer_context = context;
+	return peer_cq_init(domain, attr, cq_fid, peer_context);
+}
+
+ssize_t ofi_peer_cq_write(struct fid_cq *cq_fid, void *context, uint64_t flags,
+			size_t len, void *buf, uint64_t data, uint64_t tag,
+			fi_addr_t src)
+{
+	struct peer_cq *cq;
+
+	cq = container_of(cq_fid, struct peer_cq, util_cq.cq_fid);
+
+	return cq->peer_cq->owner_ops->write(cq->peer_cq, context,
+					  flags, len, buf, data, tag, src);
+}
```

Comment:
This function could move into the header as a static inline.

Reply:
Yes, as soon as we decide to move `peer_cq` there as well and I'm still not convinced it's necessary